drivers/adc/adc_nxp_s32_adc_sar.c

```diff
@@ -0,0 +1,458 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
+#include <Adc_Sar_Ip_HwAccess.h>
+#include <Adc_Sar_Ip.h>
+#include <Adc_Sar_Ip_Irq.h>
+
+#define ADC_CONTEXT_USES_KERNEL_TIMER
+#include "adc_context.h"
+
+#define DT_DRV_COMPAT nxp_s32_adc_sar
+LOG_MODULE_REGISTER(adc_nxp_s32_adc_sar, CONFIG_ADC_LOG_LEVEL);
+
+/* ADC internal reference voltage in mV */
+#define ADC_NXP_S32_INTERNAL_VREF 3300
+/* Convert channel of group ADC to channel of physical ADC instance */
+#define ADC_NXP_S32_GROUPCHAN_2_PHYCHAN(group, channel)	\
+						(ADC_SAR_IP_HW_REG_SIZE * group + channel)
+
+struct adc_nxp_s32_config {
+	ADC_Type *base;
+	uint8_t group_channel;
+	uint8_t callback_select;
+	Adc_Sar_Ip_ConfigType *adc_cfg;
+	void (*irq_config_func)(const struct device *dev);
+	const struct pinctrl_dev_config *pin_cfg;
+};
+
+struct adc_nxp_s32_data {
+	const struct device *dev;
```

Comment:
I'm not seeing where this pointer is needed?

Reply:
Right. I removed.